presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/ThriftModule.java

```diff
@@ -57,4 +66,12 @@ public Executor createMetadataRefreshExecutor(ThriftConnectorConfig config)
     {
         return newFixedThreadPool(config.getMetadataRefreshThreads(), daemonThreadsNamed("metadata-refresh-%s"));
     }
+
+    @Provides
+    @Singleton
+    @ForRetryDriver
+    public ScheduledExecutorService createRetryDriverScheduledExecutor(ThriftConnectorConfig config)
+    {
+        return newScheduledThreadPool(config.getMetadataRefreshThreads(), threadsNamed("thrift-retry-driver-%s"));
```

Comment:
Also, who is responsible for closing this pool?

Reply:
It's a singleton just like the thread pool for `MetadataRefreshingExecutor`.